### PR TITLE
FIX: Handle no-offsets in collection datalim (alternative)

### PR DESCRIPTION
**Pull request by greglucas**
## PR Summary

Alternative to #22945

This keeps the None-default case in `self._offsets` and then transforms to the zeros for default math in the `get_offsets()` method. This is necessary to handle the (0, 0) offset case that is desired and distinguishing that from the initial empty offset of None. In particular, this is necessary for the non-transData IdentityTransform passed in during tight_layout calculations.

closes https://github.com/matplotlib/matplotlib/issues/22921

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/collections.py [hunk 1]
```diff
@@ -193,9 +192,7 @@ def __init__(self,
         else:
             self._joinstyle = None
 
-        # default to zeros
-        self._offsets = np.zeros((1, 2))
-
+        self._offsets = offsets
```
Comment (oscargus) [inline lib/matplotlib/collections.py:195]: Could one move this to after the if-clause? (And more the identical line out of it.)

**Review by oscargus — APPROVED** [review]
Didn't know there was a `get_offset` method already. Then I think this is probably a slightly better approach.

File: lib/matplotlib/collections.py (hunk 1 above)
Comment (greglucas) [reply lib/matplotlib/collections.py]: Yep, good call, updated.

**Review by timhoffm — APPROVED** [review]

**Comment by greglucas** [comment]
@meeseeksdev backport to v3.5.x

**Pull request merged**